core/src/test/scala/pickling/run/runtime-instantiation-static-picklers.scala

```diff
@@ -0,0 +1,219 @@
+package scala.pickling.runtime
+
+import org.scalatest.FunSuite
+import scala.pickling._, scala.pickling.Defaults._, json._
+
+case class Container[T](value: T)
+
+object RuntimeHelper {
+  /* Instantiate a class for an already generated `Unpickler` */
+  def getInstance[T](className: String): T = {
+    try {
+      val clazz = Class.forName(className)
+      println(s"Create instance of $clazz")
+      clazz.newInstance().asInstanceOf[T]
+    } catch {
+      case ex: Throwable =>
+        scala.concurrent.util.Unsafe.instance.allocateInstance(
+          Class.forName(className)
+        ).asInstanceOf[T]
+    }
+  }
+}
+
+class StaticJavaRuntime extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+
+  test("runtime instantiated pickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = implicitly[Pickler[Container[List[Int]]]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
```

Comment:
So I guess that I'd need hybrid-static picklers since they match my exact use case. The thing is that I can't register static picklers for runtime lookup because that registering is actually done at runtime (unless you're inlining the code responsible of that in the initialization of all the scala pickling machinery using macros). As it's done at runtime, the server (which is an independent app than the client) will never execute that code even though they share the same jars. Is there any way to actually register those static picklers automatically without me doing anything?